cloudify_azure/resources/base.py

```diff
@@ -362,12 +362,35 @@ def delete(self, name):
                 retry_after=self.get_retry_after(headers))
         # HTTP 400 (BAD_REQUEST) - We're sending bad data
         elif res.status_code == httplib.BAD_REQUEST:
+            # Sometime it takes longer time to fully delete a parent node, so
+            # lets wait for sometime to give it a chance to be fully deleted
+            # This check will cover: [NicInUse, InUseSubnetCannotBeDeleted,
+            # InUseNetworkSecurityGroupCannotBeDeleted]
+            if "InUse" in res.json()['error']['code']:
```

Comment:
@GhalebKhaled Are we sure that each BAD request should contains an object called ``error`` and the ``code`` key will be exists ?

Reply:
I think so -- azure docs hard follow up with
https://docs.microsoft.com/en-us/rest/api/storageservices/status-and-error-codes2